### PR TITLE
cli: add --json flag to status command

### PR DESCRIPTION
**Pull request by ecrousseau**
Adds the option of JSON output to the status command, which would be useful for me, and possibly others.

### CODE REVIEW

File: cmd/status.go [hunk 1]
```diff
@@ -16,6 +17,9 @@ var statusCmd = &cobra.Command{
 	Long:  `Show the status of Colima`,
 	Args:  cobra.MaximumNArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
+		if statusCmdArgs.output == "json" {
```
Comment (abiosoft) [inline cmd/status.go:20]:
```suggestion
		if statusCmdArgs.json {
```

**Comment by abiosoft** [comment]
Thanks for this :)

**Comment by ecrousseau** [comment]
I've pushed an update that avoids changing the interface, as you suggested. My thinking on using `--output json` rather than just `--json` was that it leaves room to add other output formats in the future. Easy to change it to a boolean flag if you'd prefer, though.

**Comment by abiosoft** [comment]
> My thinking on using --output json rather than just --json was that it leaves room to add other output formats in the future.

Considering that `--json` is already used for the `colima list` command, I would prefer sticking to `--json` for consistency.

**Comment by ecrousseau** [comment]
Ah I did not realise there was an existing pattern - completely agree it should stay consistent!

File: app/app.go [hunk 2]
```diff
@@ -292,54 +292,105 @@ func (c colimaApp) SSH(args ...string) error {
...
-	log.Println("arch:", c.guest.Arch())
-	log.Println("runtime:", currentRuntime)
-	if conf.MountType != "" {
-		log.Println("mountType:", conf.MountType)
+	status.Arch = fmt.Sprintf("%v", c.guest.Arch())
+	status.Runtime = currentRuntime
+	status.MountType = conf.MountType
+	ipAddress := limautil.IPAddress(config.CurrentProfile().ID)
+	if ipAddress != "127.0.0.1" {
+		status.IPAddress = limautil.IPAddress(config.CurrentProfile().ID)
```
Comment (abiosoft) [inline app/app.go:332]: There is no need to do the lookup again.

```suggestion
		status.IPAddress = ipAddress
```

**Review by abiosoft — DISMISSED** [review]

File: app/app.go [hunk 3]
```diff
@@ -30,7 +30,7 @@ type App interface {
 	Stop(force bool) error
 	Delete() error
 	SSH(args ...string) error
-	Status(extended bool) error
+	Status(extended bool, jsonOutput bool) error
```
Comment (abiosoft) [inline app/app.go:33]:
```suggestion
	Status(extended bool, json bool) error
```

**Review by abiosoft — DISMISSED** [review]

File: cmd/status.go [hunk 4]
```diff
@@ -16,12 +17,16 @@ var statusCmd = &cobra.Command{
 	Long:  `Show the status of Colima`,
 	Args:  cobra.MaximumNArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
-		return newApp().Status(statusCmdArgs.extended)
+		if statusCmdArgs.json {
+			return newApp().Status(statusCmdArgs.extended, true)
+		}
+		return newApp().Status(statusCmdArgs.extended, false)
```
Comment (abiosoft) [inline cmd/status.go:23]:
```suggestion
		return newApp().Status(statusCmdArgs.extended, statusCmdArgs.json)
```

File: app/app.go [hunk 5]
```diff
@@ -292,54 +292,105 @@ func (c colimaApp) SSH(args ...string) error {
...
-
-	// ip address
-	if ipAddress := limautil.IPAddress(config.CurrentProfile().ID); ipAddress != "127.0.0.1" {
-		log.Println("address:", ipAddress)
-	}
-
-	// docker socket
 	if currentRuntime == docker.Name {
-		log.Println("socket:", "unix://"+docker.HostSocketFile())
+		status.DockerSocket = fmt.Sprintf("unix://%s", docker.HostSocketFile())
```
Comment (abiosoft) [inline app/app.go:335]:
```suggestion
		status.DockerSocket = "unix://" + docker.HostSocketFile()
```

File: app/app.go [hunk 6]
```diff
@@ -292,54 +292,105 @@ func (c colimaApp) SSH(args ...string) error {
...
+		}
+
+		// ip address
+		if status.IPAddress != "" {
+			log.Println("address:", status.IPAddress)
+		}
+
+		// docker socket
+		if status.DockerSocket != "" {
+			log.Println("socket:", "unix://"+status.DockerSocket)
```
Comment (abiosoft) [inline app/app.go:373]:
```suggestion
			log.Println("socket:", status.DockerSocket)
```

File: app/app.go [hunk 7]
```diff
@@ -292,54 +292,105 @@ func (c colimaApp) SSH(args ...string) error {
 	return guest.SSH(workDir, args...)
 }
 
-func (c colimaApp) Status(extended bool) error {
+type status struct {
```
Comment (abiosoft) [inline app/app.go:295]:
```suggestion
type statusInfo struct {
```

File: app/app.go [hunk 8]
```diff
@@ -292,54 +292,105 @@ func (c colimaApp) SSH(args ...string) error {
...
 	}
 
 	currentRuntime, err := c.currentRuntime(ctx)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	driver := "QEMU"
+	var status status
```
Comment (abiosoft) [inline app/app.go:320]:
```suggestion
	var status statusInfo
```

File: app/app.go [hunk 9]
```diff
@@ -292,54 +292,105 @@ func (c colimaApp) SSH(args ...string) error {
...
-		driver = conf.DriverLabel()
+		status.Driver = conf.DriverLabel()
 	}
-
-	log.Println(config.CurrentProfile().DisplayName, "is running using", driver)
-	log.Println("arch:", c.guest.Arch())
-	log.Println("runtime:", currentRuntime)
-	if conf.MountType != "" {
-		log.Println("mountType:", conf.MountType)
+	status.Arch = fmt.Sprintf("%v", c.guest.Arch())
```
Comment (abiosoft) [inline app/app.go:327]:
```suggestion
	status.Arch = string(c.guest.Arch())
```

**Review by abiosoft — DISMISSED** [review]

File: app/app.go [hunk 10]
```diff
@@ -292,54 +292,105 @@ func (c colimaApp) SSH(args ...string) error {
...
+	MountType    string `json:"mount_type"`
+	IPAddress    string `json:"ip_address"`
+	DockerSocket string `json:"docker_socket"`
+	Kubernetes   bool   `json:"kubernetes"`
+	CPU          int    `json:"cpu"`
+	Memory       int64  `json:"memory"`
+	Disk         int64  `json:"disk"`
+}
+
+func (c colimaApp) getStatus() (*status, error) {
```
Comment (abiosoft) [inline app/app.go:309]:
```suggestion
func (c colimaApp) getStatus() (status statusInfo, err error) {
```

File: app/app.go [hunk 11]
```diff
@@ -292,54 +292,105 @@ func (c colimaApp) SSH(args ...string) error {
...
 	ctx := context.Background()
 	if !c.guest.Running(ctx) {
-		return fmt.Errorf("%s is not running", config.CurrentProfile().DisplayName)
+		return nil, fmt.Errorf("%s is not running", config.CurrentProfile().DisplayName)
```
Comment (abiosoft) [inline app/app.go:312]:
```suggestion
		return status, fmt.Errorf("%s is not running", config.CurrentProfile().DisplayName)
```

File: app/app.go [hunk 12]
```diff
@@ -292,54 +292,105 @@ func (c colimaApp) SSH(args ...string) error {
...
 	}
 
 	currentRuntime, err := c.currentRuntime(ctx)
 	if err != nil {
-		return err
+		return nil, err
```
Comment (abiosoft) [inline app/app.go:317]:
```suggestion
		return status, err
```

File: app/app.go [hunk 13]
```diff
@@ -292,54 +292,105 @@ func (c colimaApp) SSH(args ...string) error {
...
 	if k, err := c.Kubernetes(); err == nil && k.Running(ctx) {
-		log.Println("kubernetes: enabled")
+		status.Kubernetes = true
+	}
+	if inst, err := limautil.Instance(); err == nil {
+		status.CPU = inst.CPU
+		status.Memory = inst.Memory
+		status.Disk = inst.Disk
 	}
+	return &status, nil
```
Comment (abiosoft) [inline app/app.go:345]:
```suggestion
	return status, nil
```

File: app/app.go [hunk 14]
```diff
@@ -292,54 +292,105 @@ func (c colimaApp) SSH(args ...string) error {
...
-		}
+func (c colimaApp) Status(extended bool, jsonOutput bool) error {
+	status, err := c.getStatus()
+	if err != nil {
+		return err
 	}
 
+	if jsonOutput {
+		b, _ := json.Marshal(status)
+		fmt.Println(string(b))
```
Comment (abiosoft) [inline app/app.go:356]:
```suggestion
		if err := json.NewEncoder(os.Stdout).Encode(status); err != nil {
			return fmt.Errorf("error encoding status as json: %w", err)
		}
```

File: app/app.go [hunk 15]
```diff
@@ -292,54 +292,106 @@ func (c colimaApp) SSH(args ...string) error {
...
 
 	currentRuntime, err := c.currentRuntime(ctx)
 	if err != nil {
-		return err
+		return status, err
 	}
 
-	driver := "QEMU"
+	var status statusInfo
+	status.DisplayName = config.CurrentProfile().DisplayName
```
Comment (abiosoft) [inline app/app.go:321]:
```suggestion
	status.DisplayName = config.CurrentProfile().DisplayName
```

**Review by abiosoft — APPROVED** [review]

**Comment by abiosoft** [comment]
Once again, thanks.

**Pull request merged**